lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/HeartBeatManager.cs

```diff
@@ -126,6 +128,18 @@ public void Send(EvaluatorHeartbeatProto evaluatorHeartbeatProto)
         {
             lock (_queuedHeartbeats)
             {
+                // Do not send a heartbeat if Evaluator has already signaled that it was done.
+                if (_isCompletedHeartbeatQueued)
```

Comment:
Can you just check (IsEvaluatorStateCompleted(evaluatorHeartbeatProto.evaluator_status.state) instead of introduce a variable _isCompletedHeartbeatQueued?